### DISCUSSION TITLE
yolov5 corner case

### DISCUSSION THREAD

**Discussion by yuangpeng** [open-mmlab/mmyolo#619] [Q&A] [answered]
<img width="913" alt="image" src="https://user-images.githubusercontent.com/57125678/222636177-11aaff5b-c9a5-47e3-9395-4f27c8152b97.png">
I would like to ask what the corner case refers to here

**Comment by hhaAndroid** [comment, marked as answer]
@yuangpeng  有很多，下面列几个典型的

- 自动 bs 计算，会在训练前自动推断你的最佳 bs
- 自动权重衰减缩放
- 自动 anchor 优化
- 自动 Loss 权重缩放
- 如果提供的数据集路径不存在，自动进行数据集下载
- 当用户的 bs 小于 64 时候自动切换为梯度累加模式
- 推理和验证阶段自动 fuse half 转换，加快计算

**Reply by yuangpeng** [reply]
谢谢
